reactive/bundleservice_charm.py

```diff
@@ -0,0 +1,43 @@
+from charms.reactive import (
+    hook,
+    set_state,
+    when,
+    when_not,
+)
+from charmhelpers.core import (
+    hookenv,
+    host,
+)
+
+
+@when('nrpe-external-master.available')
+def setup_nagios(nagios):
+    config = hookenv.config()
+    unit_name = hookenv.local_unit()
+    nagios.add_check(
+        ['/usr/lib/nagios/plugins/check_http',
+         '-I', '127.0.0.1', '-p', str(config['listen-port']),
+         '-e', " 404 Not Found", '-u', '/'],
+        name="check_http",
+        description="Verify bundleservice is running",
+        context=config["nagios_context"],
+        unit=unit_name,
+    )
```

Comment:
Based on the recent conversation between Jeff and Marco, I think this setup_nagios method is going to be called every loop run. What will nagios.add_check do, add a new entry every time?

Regardless this method needs a @when_not('nrpe.configured') or similar. You'll also need to decide how to remove that state, I guess in config_changed.